client/my-sites/stats/stats-card-upsell/index.tsx

```diff
@@ -1,60 +1,34 @@
+import config from '@automattic/calypso-config';
 import { translate } from 'i18n-calypso';
-import { useDispatch } from 'react-redux';
-import { preventWidows } from 'calypso/lib/formatting';
-import { toggleUpsellModal } from 'calypso/state/stats/paid-stats-upsell/actions';
-import {
-	STATS_FEATURE_DATE_CONTROL,
-	STAT_TYPE_CLICKS,
-	STAT_TYPE_REFERRERS,
-	STAT_TYPE_SEARCH_TERMS,
-	STAT_TYPE_TOP_AUTHORS,
-} from '../constants';
-import StatsCardUpsellOverlay from './stats-card-upsell-overlay';
+import { useSelector } from 'calypso/state';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import StatsCardUpsellJetpack from './stats-card-upsell-jetpack';
+import StatsCardUpsellWPCOM from './stats-card-upsell-wpcom';
 
-import './style.scss';
-
-interface Props {
+export interface Props {
 	className: string;
 	statType: string;
 	siteId: number;
 	buttonLabel?: string;
 }
 
-const getUpsellCopy = ( statType: string ) => {
-	switch ( statType ) {
-		case STAT_TYPE_REFERRERS:
-			return translate(
-				'Find out where your visitors come from to optimize your content strategy.'
-			);
-		case STAT_TYPE_CLICKS:
-			return translate(
-				'Learn what external links your visitors click on your site to reveal their areas of interest.'
-			);
-		case STAT_TYPE_SEARCH_TERMS:
-			return translate( 'Discover the terms and phrases your visitors use to find your site.' );
-		case STAT_TYPE_TOP_AUTHORS:
-			return translate( 'Identify your audience’s favorite writers and perspectives.' );
-		case STATS_FEATURE_DATE_CONTROL:
-			return translate( 'Compare different time periods to analyze your site’s growth.' );
-		default:
-			return translate( 'Upgrade your plan to unlock Jetpack Stats.' );
-	}
-};
-
 const StatsCardUpsell: React.FC< Props > = ( { className, statType, siteId, buttonLabel } ) => {
-	const dispatch = useDispatch();
+	const isSiteJetpackNotAtomic = useSelector( ( state ) =>
+		isJetpackSite( state, siteId, { treatAtomicAsJetpackSite: false } )
+	);
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
 
-	const onClick = ( event: React.MouseEvent< HTMLButtonElement, MouseEvent > ) => {
-		event.preventDefault();
-		dispatch( toggleUpsellModal( siteId, statType ) );
-	};
+	let UpsellComponent = StatsCardUpsellWPCOM;
+	if ( isSiteJetpackNotAtomic || isOdysseyStats ) {
```

Comment:
Is this logic correct?

`isSiteJetpackNotAtomic` should see `StatsCardUpsellJetpack`, but `isOdysseyStats` can soon include both Simple and Atomic sites.

I thought it would have been the following:
- Self-hosted Jetpack -> `StatsCardUpsellJetpack`
- Simple and Atomic -> `StatsCardUpsellWPCOM`

```suggestion
	if ( isSiteJetpackNotAtomic ) {
```

Reply:
Good point! I think you are right. Addressed in 0e9da2b9e6e.